### ISSUE TITLE
JPMS

### ISSUE THREAD

**Issue by palexdev** [shrinkwrap/resolver#353]
Any plan on making this compatible with modular projects?
I'm having a very hard time here trying to make it work with Gradle's extraJavaModuleInfo

**Comment by palexdev** [comment]
> Do you have a specific issue? I use resolver with modular apps all the time

When I add this `requires shrinkwrap.resolver.api.maven;` to the module-info I get this error: `error: module not found: shrinkwrap.resolver.api.maven`

**Comment by palexdev** [comment]
It's not a matter or names
Modules support is missing. I wonder if something like [this](https://github.com/apache/maven-resolver/pulls?q=is%3Apr+jpms) could be done here too

**Comment by lprimak** [comment]
By default, automatic module naming takes place, based on the dependency name.
See https://stackoverflow.com/questions/46741907/what-is-an-automatic-module

While specific module support is nice, it's unnecessary to support JPMS clients

**Comment by palexdev** [comment]
Yeah the automatic module name should work but for some reason it does not
I'm migrating to a modular library for now

**Comment by lprimak** [comment]
I *think* we are trying to say the same thing...
if you get the name right it should work

**Comment by palexdev** [comment]
Here's a [MRE](https://www.dropbox.com/scl/fi/6mgnhnlnxgyk93fcudwjs/shrinkwrap-mre.zip?rlkey=9hqvg6lj2ndxl0ujs8nx7e6b1&st=o5cucc0l&dl=0)

**Comment by lprimak** [comment]
Maven please... I don't use gradle :)

**Comment by palexdev** [comment]
> Maven please... I don't use gradle :)

Not even IntelliJ?

**Comment by lprimak** [comment]
This is a debugging / troubleshooting / exploration task. I really don't want to learn a new build tool :)
IntelliJ doesn't do well with automatic modules, as I had problems building with IntelliJ but it would build correctly in maven command line.

Maybe that's the root cause of your problem.

**Comment by palexdev** [comment]
So, at this point, I don't really know if it's a Gradle issue or what: [Documentation](https://docs.gradle.org/8.10/userguide/java_library_plugin.html#sec:java_library_modular).

One thing is certain: it's disappointing to see such a useful library in _2024_ still targeting _JDK 8_ and lacking proper module support. JDK 9 was released in 2017, with JDK 11 following the next year — it's time to move on, folks.

Finally, as mentioned before, I’m transitioning to another library with proper modules support. Feel free to close this issue or keep it open as a reminder for future improvement.

**Comment by lprimak** [comment]
ShrinkWrap resolver is tested up to JDK 17 now, so it fully supports JDK 17. JDK 8 is coincidental / legacy as you mentioned, and it's still supported because there is no downside right now.

The library is primarily used for testing, and testing realm isn't really modularized often, so your issue here is really an outlier.

Will keep the issue open, and it's good practice to put in the MANIFEST.MF entry for automatic module support.
However, I really don't think this is in any way related to the problem your are facing.
